hummingbot/connector/exchange/bitstamp/bitstamp_exchange.py

```diff
@@ -0,0 +1,530 @@
+        try:
+            event_data = event_message.get("data", {})
+            client_order_id = str(event_data.get("client_order_id"))
+            order: InFlightOrder = self._order_tracker.all_fillable_orders.get(client_order_id)
+            if order is None:
+                self.logger().debug(f"Received event for unknown order ID: {event_message}")
+                return
+
+            # Determine the new state of the order
+            new_state = OrderState.OPEN
+            if event == CONSTANTS.USER_ORDER_CHANGED and Decimal(event_data["amount_traded"]) > 0:
+                new_state = OrderState.PARTIALLY_FILLED
+            elif event == CONSTANTS.USER_ORDER_DELETED:
+                new_state = OrderState.FILLED if event_data["amount"] == 0 else OrderState.CANCELED
+
+            order_update = OrderUpdate(
+                client_order_id=order.client_order_id,
+                exchange_order_id=order.exchange_order_id,
+                trading_pair=order.trading_pair,
+                update_timestamp=float(event_data["datetime"]),
+                new_state=new_state,
+            )
+            self._order_tracker.process_order_update(order_update)
+        except Exception as e:
+            raise ValueError(f"Error parsing the user stream order event {event_message}: {e}")
+
+    async def _all_trade_updates_for_order(self, order: InFlightOrder) -> List[TradeUpdate]:
+        all_fills_response = await self._api_post(
+            path_url=CONSTANTS.ORDER_STATUS_URL,
+            data={
+                "client_order_id": order.client_order_id,
+                "omit_transactions": "false",
+            },
+            is_auth_required=True
+        )
+
+        exchange_order_id = await order.get_exchange_order_id()
+        trade_updates = []
+        for trade in all_fills_response.get("transactions", []):
+            fee = TradeFeeBase.new_spot_fee(
+                fee_schema=self.trade_fee_schema(),
+                trade_type=order.trade_type,
+                flat_fees=[TokenAmount(amount=Decimal(trade["fee"]), token=order.quote_asset)]
+            )
+            trade_update = TradeUpdate(
+                trade_id=str(trade["tid"]),
+                client_order_id=order.client_order_id,
+                exchange_order_id=exchange_order_id,
+                trading_pair=order.trading_pair,
+                fee=fee,
+                fill_base_amount=Decimal(trade[order.base_asset.lower()]),
+                fill_quote_amount=Decimal(trade[order.quote_asset.lower()]),
+                fill_price=Decimal(trade["price"]),
+                fill_timestamp=datetime.fromisoformat(trade["datetime"]).timestamp(),
+            )
+            trade_updates.append(trade_update)
+
+        return trade_updates
+
+    async def _request_order_status(self, tracked_order: InFlightOrder) -> OrderUpdate:
+        updated_order_data = await self._api_post(
```

Comment:
this is what I think that we can improve later on and don't ask for each order, instead use the endpoint and collect the updates from the orders returned and if one is not appearing then do the custom request.
not to implement right now but what do you think?